Scripts/UtilityScripts/AzTSUtility.ps1

```diff
@@ -0,0 +1,1617 @@
+    .PARAMETER DataFactoryName
+    The Data Factory name.
+    .INPUTS
+    None
+    .OUTPUTS
+    For each Data Flow, text with the Data Flow name and its script lines, which begin with parameters and default parameter values.
+    .EXAMPLE
+    PS> Get-DataFactoryV2DataFlows -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -DataFactoryName "MyDataFactoryName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+      [Parameter(Mandatory=$true)]
+      [string]
+      $SubscriptionId,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $ResourceGroupName,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $DataFactoryName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $dataflows = Get-AzDataFactoryV2DataFlow -ResourceGroupName $ResourceGroupName -DataFactoryName $DataFactoryName
+
+  foreach ($dataflow in $dataflows)
+  {
+    Write-Information -InformationAction Continue -MessageData ("Dataflow: " + $dataflow.Name)
+    Write-Information -InformationAction Continue -MessageData "Dataflow Script Lines:"
+    $dataflows.Properties.ScriptLines
+    Write-Information -InformationAction Continue -MessageData ""
+  }
+
+}
+
+function Get-DataFactoryV2DataSets()
+{
+  <#
+    .SYNOPSIS
+    This command lists dataset names and parameters for datasets that have at least one parameter.
+    .DESCRIPTION
+    This command lists dataset names and parameters for datasets that have at least one parameter. This is to review parameter values to try and locate parameters that may contain strings identified by the control as plaintext secrets.
+    .PARAMETER SubscriptionId
+    The Azure subscription ID containing the Data Factory.
+    .PARAMETER ResourceGroupName
+    The Resource Group name containing the Data Factory.
+    .PARAMETER DataFactoryName
+    The Data Factory name.
+    .INPUTS
+    None
+    .OUTPUTS
+    For each Dataset, text with the Dataset name and its parameter names and values.
+    .EXAMPLE
+    PS> Get-DataFactoryV2DataSets -SubscriptionId "00000000-xxxx-0000-xxxx-000000000000" -ResourceGroupName "MyResourceGroupName" -DataFactoryName "MyDataFactoryName"
+    .LINK
+    None
+  #>
+
+  [CmdletBinding()]
+  param (
+      [Parameter(Mandatory=$true)]
+      [string]
+      $SubscriptionId,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $ResourceGroupName,
+      [Parameter(Mandatory=$true)]
+      [string]
+      $DataFactoryName
+  )
+
+  $profile = Set-AzContext -Subscription $SubscriptionId
+
+  $datasets = Get-AzDataFactoryV2Dataset -ResourceGroupName $ResourceGroupName -DataFactoryName $DataFactoryName
+
+  foreach ($dataset in $datasets)
+  {
+    Write-Information -InformationAction Continue -MessageData ("Dataset: " + $dataset.Name)
+    Write-Information -InformationAction Continue -MessageData "Parameter Names and Values:"
+    foreach ( $h in $dataset.Properties.Parameters.GetEnumerator() )
```

Comment:
Plz validate if null check required here, I got null ref exception for one of the ADF, 

You cannot call a method on a null-valued expression.
At line:330 char:21
+     foreach ( $h in $dataset.Properties.Parameters.GetEnumerator() )

Reply:
Thanks. Added null check for Parameters, which mitigates this defect.